app/src/main/java/com/duckduckgo/app/feedback/ui/SurveyActivity.kt

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.feedback.ui
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.webkit.*
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.Observer
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.feedback.model.Survey
+import com.duckduckgo.app.feedback.ui.SurveyViewModel.Command
+import com.duckduckgo.app.feedback.ui.SurveyViewModel.Command.Close
+import com.duckduckgo.app.feedback.ui.SurveyViewModel.Command.LoadSurvey
+import com.duckduckgo.app.global.DuckDuckGoActivity
+import com.duckduckgo.app.global.view.gone
+import com.duckduckgo.app.global.view.show
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.app.statistics.pixels.Pixel.PixelName.*
+import kotlinx.android.synthetic.main.activity_user_survey.*
+import javax.inject.Inject
+
+
+class SurveyActivity : DuckDuckGoActivity() {
+
+    private val viewModel: SurveyViewModel by bindViewModel()
+
+    @Inject
+    lateinit var pixel: Pixel
+
+    @SuppressLint("SetJavaScriptEnabled")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_user_survey)
+        configureListeners()
+
+        webView.settings.javaScriptEnabled = true
+        webView.setBackgroundColor(ContextCompat.getColor(this, R.color.cornflowerBlue))
+        webView.webViewClient = SurveyWebViewClient()
+
+        if (savedInstanceState == null) {
+            configureObservers()
+            consumeIntentExtra()
+        }
+    }
+
+    private fun consumeIntentExtra() {
+        val survey = intent.getSerializableExtra(SURVEY_EXTRA) as Survey
+        viewModel.start(survey)
+    }
+
+    private fun configureListeners() {
+        dismissButton.setOnClickListener {
```

Comment:
should the X button and the back button do the same thing? or are they different deliberately?

Reply:
Good point, they should be different. I've tweaked them to make this work better and more readable